src/hooks/core.ts

```diff
@@ -0,0 +1,117 @@
+import { createRoot } from "../render";
+import { JSXNode, SetStateAction } from "../type";
+
+interface Effect {
+  callback: () => void | (() => void);
+  dependencies: any[] | undefined;
```

Comment:
요런 경우에 아래처럼 해도 괜찮을 것 같아요!


```suggestion
  dependencies?: any[];
```